packages/block-editor/src/components/page-template-picker/with-page-template-picker.native.js

```diff
@@ -13,8 +13,15 @@ import {
 
 const __experimentalWithPageTemplatePicker = createHigherOrderComponent(
 	( WrappedComponent ) => ( props ) => {
-		const isTemplatePickerVisible = __experimentalUsePageTemplatePickerVisible();
-		const isTemplatePickerAvailable = __experimentalUsePageTemplatePickerAvailable();
+		const isTemplatePickerEnabled = ! (
+			props.capabilities.modalLayoutPicker ?? false
+		);
+		const isTemplatePickerVisible =
+			__experimentalUsePageTemplatePickerVisible() &&
+			isTemplatePickerEnabled;
```

Comment:
What do you think if we switch the order to:

```suggestion
			isTemplatePickerEnabled &&
			__experimentalUsePageTemplatePickerVisible();
```

This way we avoid executing `__experimentalUsePageTemplatePickerVisible` if `isTemplatePickerEnabled` is false.